ttg/ttg/util/device.h

```diff
@@ -0,0 +1,13 @@
+#ifndef TTG_UTILS_DEVICE_H
+#define TTG_UTILS_DEVICE_H
```

Comment:
This is not util functionality, but provided to the user. Should be one level up. I think we should put all device-related functions and objects into the `ttg::device::` namespace.

Reply:
Fine with me, I'll promote.

Reply:
See e7fc76987b84b7bd793b837d6e34dd2407aa4f53